src/master/master.ts

```diff
@@ -296,10 +296,16 @@ export class Master {
       };
     });
 
+    const { deltalog, ...stateWithoutDeltalog } = state;
+
     if (IsSynchronous(this.storageAPI)) {
-      this.storageAPI.setState(key, state);
+      this.storageAPI.setState(key, stateWithoutDeltalog);
```

Comment:
If the storage implementation wants to perform this update in one atomic operation (using a transaction at the database layer), splitting this into two calls will make it hard for it to do so.  Sorry I didn't catch this earlier.

How about we pass `deltalog` as an additional parameter to the `setState` call?  That way it's a bit more explicit than just piggybacking on the `state` object.

Reply:
Hmm, OK, yes that does make sense. OK I’ll revert it back to more or less what it was, but with `deltalog` as an optional argument :-)

Reply:
Done in 063721f.